main.go

```diff
@@ -3,16 +3,41 @@ package main
 import (
 	"fmt"
 	"log"
+	"os"
+	"slices"
 )
 
 const gomodName = "go.mod"
 
 func main() {
+	if err := run(); err != nil {
```

Comment:
I find having a `run()` function that returns an error makes for easier error handling.

We then have a single place that logs out the error - and can use idiomatic go conventions for returning the error in the `run` function